src/sentry/replays/usecases/reader.py

```diff
@@ -253,22 +253,23 @@ def download_segments(segments: List[RecordingSegmentStorageMeta]) -> Iterator[b
         download_segment, transaction=transaction, current_hub=sentry_sdk.Hub.current
     )
 
-    # Map all of the segments to a worker process for download.
-    with ThreadPoolExecutor(max_workers=4) as exe:
-        results = exe.map(download_segment_with_fixed_args, segments)
-
+    # use a threadpool to download each segment from storage.
+    # stream the downloaded segments as they come in, in order.
     yield b"["
+    with ThreadPoolExecutor(max_workers=4) as exe:
```

Comment:
Double check this value is correct after https://github.com/getsentry/sentry/pull/47568 is merged.